streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -178,6 +184,67 @@ public void register(final StateStore store,
         stores.put(store.name(), store);
     }
 
+    public void reinitializeStateStoresForPartitions(final Collection<TopicPartition> partitions,
```

Comment:
Just confirming my understanding: "reinitialize" means "delete the files and restore from changelog"?

This seems un-symmetrically named with "initialize", which means "open the stores on disk".

Or have I missed something?

Reply:
> Just confirming my understanding: "reinitialize" means "delete the files and restore from changelog"?

Yes. And "initialize" is not just "open the store on disk" but also, do a restore if required -- but normal restore in initialize does not wipe the store, and does start from offset as stored in `.checkpoint` file.